ramips: improve support for HC5962

dengqf6 opened the pull request:
- fixed switch configuration
- fixed status LED
- reorganized partitions

adrianschmutzler commented on target/linux/ramips/dts/mt7621_hiwifi_hc5962.dts:
```diff
@@ -23,14 +23,15 @@
 		};
 
-		system {
+		led_red: system {
```
Since the LEDs have "system" and "status" for names, I'd reflect that in the labels:
led_status and led_system
OR
led_status_white and led_system_red

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -337,9 +337,6 @@ define Device/hiwifi_hc5962
   DEVICE_VENDOR := HiWiFi
   DEVICE_MODEL := HC5962
   DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 kmod-usb3 wpad-basic
-  SUPPORTED_DEVICES += hc5962
```
Removing this will only prevent flashing without -F from stable releases, but flashing from recent snapshots after the ramips.sh removal will still work without -F.
However, I think this is acceptable in this case.

adrianschmutzler commented:
I've pushed 1 and 2, and will also backport them later.

adrianschmutzler commented:
Backport staging:
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/staging-19.07

I won't merge the partitions patch, so you will have wait for someone else there.

dengqf6 commented:
@blocktrron

blocktrron commented on target/linux/ramips/dts/mt7621_hiwifi_hc5962.dts:
```diff
@@ -44,6 +44,22 @@
+				reg = <0x0 0x79c0000>;
+			};
+		};
+	};
```
I'm not sure if/how badblock handling works with mtd-concat. Did you spend time with this?

If not, I'm fine with enlarging the kernel partition but would like to keep this out of here.

dengqf6 replied:
It will be handled by UBI.
cc @realmicu

dengqf6 replied:
I've splitted the commit into two. The first one enlarges the kernel partition and the second one uses mtd-concat.

The mtd-concat idea was from ad6c6361f9f85f7cf6f53558d109c82de73743a5 by @realmicu

realmicu replied:
I confirm, I've never had any problems with such mtd-concat setup on my Netgear WNDR4300 v1. 
Please be aware though that using all flash on some routers may reveal low quality of flash chip validation, something I was able to fix from u-boot prompt : https://forum.openwrt.org/t/support-for-netgear-wndr4300-v2/26099/24

blocktrron reviewed (commented):
I'm fine with enlarging the kernel. However (if possible) i would

dengqf6 commented:
@blocktrron ping

dengqf6 commented:
@adrianschmutzler @blocktrron Can you push the adjust kernel size patch now?

Commit:
ramips: increase HiWiFi HC5962 kernel partition to 4M

Increase kernel partition because 2M is insufficient for 5.4
Because the partition changes, previous version of OpenWrt cannot upgrade
to this version, and requires a new installation

Recovery to stock instruction:
1. Download stock firmware at
   http://ur.ikcd.net/HC5962-sysupgrade-20171221-b00a04d1.bin
2. Power off the router
3. Press and hold the reset button for 4~6 sec while power it back on
4. Connect a PC to router's LAN
5. Visit http://192.168.2.1 and upload the firmware

Then repeat the instruction in edae3479e64e to install OpenWrt

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
ramips: use all reserved space for HiWiFi HC5962

These stock partitons: "backup", "hw_panic", "overly", firmware_backup", "opt"
do not contain any device-specific data and can be used for /overlay, resulting in
121M space

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

jfwefh commented:
Could someone please merge this patch? It's preventing us from generating new images for this device right now.

981213 commented:
Merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Thanks!
